driver/labjack/reader_source.cpp

```diff
@@ -85,30 +90,28 @@ void labjack::ReaderSource::init() {
     auto [dev, err] = this->ctx->client->hardware.retrieveDevice(
         this->reader_config.device_key
     );
-    if (err != freighter::NIL) {
-        LOG(ERROR) << "[labjack.reader] Error retrieving device: " << err.message();
-        return;
-    }
 
-    if (dev.model == "LJM_dtT4") {
+    if (err != freighter::NIL)
+        return this->log_err("Error retrieving device: " + err.message());
+
+    if (dev.model == "LJM_dtT4")
         this->reader_config.device_type = "T4";
-    } else if (dev.model == "LJM_dtT7") {
+    else if (dev.model == "LJM_dtT7")
         this->reader_config.device_type = "T7";
-    } else if (dev.model == "LJM_dtT8") {
+    else if (dev.model == "LJM_dtT8")
         this->reader_config.device_type = "T8";
-    } else {
-        LOG(ERROR) << "[labjack.reader] Unsupported device type: " << dev.model;
-        return;
-    }
+    else
+        return this->log_err("Unsupported device type: " + dev.model);
+
     this->init_stream();
     this->init_tcs();
 }
 
 
 void labjack::ReaderSource::init_tcs() {
-    if (this->reader_config.tc_channels.empty()) {
-        return;
-    }
+    if (this->reader_config.tc_channels.empty()) return;
+    if(this->reader_config.device_type == "T4")
```

Comment:
Why don't we support TCs on a T4? Was this always the case?

Reply:
Yes. T4 does not have the hardware built in to support thermocouple channels